nbbuild/cluster.properties

```diff
@@ -420,6 +420,7 @@ nb.cluster.ide=\
         o.apache.commons.io,\
         o.apache.commons.lang,\
         o.apache.commons.logging,\
+        o.apache.servicemix.bundles.jsch,\
```

Comment:
Don't we already have jsch in c.jcraft.jsch?

Reply:
Yes we do but this version isn't OSGi compliant and there has been a change in the manifest file of JGit starting from version 4.5 that now requires this dependency to be part of an OSGi bundle: 

https://github.com/eclipse/jgit/commit/e2cb2f8afdcfc3505f0d7ab0e084415c9358139a#diff-66e59ab180153a142ed3da762d562591

And this Apache repackaged version has been made specifically for this purpose so I thought it was a good fit. 

I guess a more proper way of handling these external OSGi dependencies would be to use the ones built directly by the Eclipse team! And I've tried to do that at first but they aren't available on Maven Central (see my post on the orbit-dev mailing list: https://dev.eclipse.org/mhonarc/lists/orbit-dev/msg04952.html)! Currently they are only available through Eclipse Orbit (P2) repository: http://download.eclipse.org/tools/orbit/downloads/drops/R20180206163158/

Maybe we could use Tycho to fetch these dependencies from the Orbit repository but I'm not familiar enough with Ant and the NetBeans build system to implement that...

Reply:
Last I checked JGit was a plain JAR, not an OSGi bundle. It might have those manifest lines, but they are safely ignored.

So:

1. Is our module OSGi or NBM? Why would we use it as OSGi for such a plain library that doesn't need anything relevant from the OSGi side (services, lifecycle) and just dependencies to be met?

2. If we are an OSGi module why does NetBeans not manage to satisfy that dependency considering we bundle jsch?

Reply:
Well I don't think that they are ignored!

Because when I started working on the JGit upgrade in NetBeans I didn't touch the dependencies at first. I simply updated the JGit version of the NetBeans module, fixed the code on the NetBeans side and then I started it and I got this exception:

```java
org.osgi.framework.BundleException: The bundle "org.eclipse.jgit_4.11.0.201803080745-r [183]" could not be resolved. Reason: Missing Constraint: Import-Package: com.jcraft.jsch; version="[0.1.37,0.2.0)"
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.getResolverError(AbstractBundle.java:1332)
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.getResolutionFailureException(AbstractBundle.java:1316)
    at org.eclipse.osgi.framework.internal.core.BundleHost.startWorker(BundleHost.java:323)
    at org.eclipse.osgi.framework.internal.core.AbstractBundle.resume(AbstractBundle.java:390)
    at org.eclipse.osgi.framework.internal.core.Framework.resumeBundle(Framework.java:1184)
    at org.eclipse.osgi.framework.internal.core.StartLevelManager.resumeBundles(StartLevelManager.java:559)
    at org.eclipse.osgi.framework.internal.core.StartLevelManager.resumeBundles(StartLevelManager.java:544)
    at org.eclipse.osgi.framework.internal.core.StartLevelManager.incFWSL(StartLevelManager.java:457)
    at org.eclipse.osgi.framework.internal.core.StartLevelManager.doSetStartLevel(StartLevelManager.java:243)
    at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.internalStart(EquinoxLauncher.java:271)
    at org.eclipse.osgi.framework.internal.core.EquinoxLauncher.start(EquinoxLauncher.java:241)
    at org.eclipse.osgi.launch.Equinox.start(Equinox.java:258)
    at org.netbeans.core.netigso.Netigso.start(Netigso.java:183)
    at org.netbeans.NetigsoHandle.startFramework(NetigsoHandle.java:186)
    at org.netbeans.ModuleManager.enable(ModuleManager.java:1344)
    at org.netbeans.ModuleManager.enable(ModuleManager.java:1148)
    at org.netbeans.core.startup.ModuleList.installNew(ModuleList.java:315)
    at org.netbeans.core.startup.ModuleList.trigger(ModuleList.java:251)
    at org.netbeans.core.startup.ModuleSystem.restore(ModuleSystem.java:276)
    at org.netbeans.core.startup.Main.getModuleSystem(Main.java:156)
    at org.netbeans.core.startup.Main.getModuleSystem(Main.java:125)
    at org.netbeans.core.startup.Main.start(Main.java:282)
    at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:98)
[catch] at java.lang.Thread.run(Thread.java:748)
```

And I got the same exception for the other 2 dependencies (Servlet API and SLF4J API) after I fixed the problem for jsch! 

So these OSGi dependencies are triggered by something during the loading of the module!

Reply:
How can I test if the current NetBeans module is OSGi or NBM?

Reply:
Based on what I'm reading here: http://wiki.apidesign.org/wiki/Netigso, it seems that the NetBeans Git module is indeed **not** an OSGi bundle (OpenIDE-Module)!

But I really think that the JGit library is an OSGi bundle, otherwise why would I get the exception above where it clearly shows that the Equinox container is being started by the NetBeans bridge (Netigso) and that it's trying to resolve JGit's dependencies?